compiler/hash-lower/src/build/ty.rs

```diff
@@ -102,63 +112,83 @@ impl<'tcx> BodyBuilder<'tcx> {
+                        );
+
+                        // Convert the `to_ty` into an IR type and
+                        let ty = self.ty_id_from_tir_ty(to_ty);
+
+                        FnCallTermKind::Cast(value, ty)
+                    }
+                    TirIntrinsic::CondBinOp => {
+                        let (op, lhs, rhs) = (
+                            args.at(1).unwrap().borrow().value,
+                            args.at(2).unwrap().borrow().value,
+                            args.at(3).unwrap().borrow().value,
+                        );
+
+                        let op = CondBinOp::try_from(
+                            try_use_term_as_integer_lit::<_, u8>(self.env(), op).unwrap(),
+                        )
+                        .unwrap();
+                        FnCallTermKind::BinaryOp(op.into(), lhs, rhs)
+                    }
+                    TirIntrinsic::ShortCircuitingBoolOp => {
+                        let (op, lhs, rhs) = (
+                            args.at(1).unwrap().borrow().value,
+                            args.at(2).unwrap().borrow().value,
+                            args.at(3).unwrap().borrow().value,
+                        );
+
+                        let op = ShortCircuitingBoolOp::try_from(
+                            try_use_term_as_integer_lit::<_, u8>(self.env(), op).unwrap(),
+                        )
+                        .unwrap();
+
+                        FnCallTermKind::LogicalBinOp(op.into(), lhs, rhs)
+                    }
+                    TirIntrinsic::BinOp => {
+                        let (op, lhs, rhs) = (
+                            args.at(1).unwrap().borrow().value,
+                            args.at(2).unwrap().borrow().value,
+                            args.at(3).unwrap().borrow().value,
+                        );
+
+                        let op = BinOp::try_from(
+                            try_use_term_as_integer_lit::<_, u8>(self.env(), op).unwrap(),
+                        )
+                        .unwrap();
+                        FnCallTermKind::BinaryOp(op.into(), lhs, rhs)
+                    }
+                    TirIntrinsic::UnOp => {
+                        let (op, subject) = (
+                            args.at(1).unwrap().borrow().value,
+                            args.at(2).unwrap().borrow().value,
+                        );
+
+                        // Parse the operator from the starting term.
+                        let parsed_op = UnOp::try_from(
+                            try_use_term_as_integer_lit::<_, u8>(self.env(), op).unwrap(),
+                        )
+                        .unwrap();
+
+                        FnCallTermKind::UnaryOp(parsed_op.into(), subject)
+                    }
+                    TirIntrinsic::SizeOf
+                    | TirIntrinsic::AlignOf
+                    | TirIntrinsic::PtrOffset
+                    | TirIntrinsic::Eval
```

Comment:
Perhaps `Eval`, `UserError` and `DebugPrint` should panic here since they aren't meant to be "lowered"?